crates/ruff_linter/src/rules/ruff/rules/sequence_sorting.rs

```diff
@@ -0,0 +1,818 @@
+            Self::List => ("[", "]"),
+            Self::Set => ("{", "}"),
+            Self::Tuple(ast_node) => {
+                if ast_node.is_parenthesized(source) {
+                    ("(", ")")
+                } else {
+                    ("", "")
+                }
+            }
+        }
+    }
+
+    fn opening_token_for_multiline_definition(&self) -> Tok {
+        match self {
+            Self::List => Tok::Lsqb,
+            Self::Set => Tok::Lbrace,
+            Self::Tuple(_) => Tok::Lpar,
+        }
+    }
+
+    fn closing_token_for_multiline_definition(&self) -> Tok {
+        match self {
+            Self::List => Tok::Rsqb,
+            Self::Set => Tok::Rbrace,
+            Self::Tuple(_) => Tok::Rpar,
+        }
+    }
+}
+
+/// An enumeration of the various kinds of
+/// [display literals](https://docs.python.org/3/reference/expressions.html#displays-for-lists-sets-and-dictionaries)
+/// Python provides for builtin containers.
+#[derive(Debug, is_macro::Is)]
+pub(super) enum DisplayKind<'a> {
+    Sequence(SequenceKind<'a>),
+    Dict { values: &'a Vec<ast::Expr> },
+}
+
+/// Create a string representing a fixed-up single-line
+/// definition of `__all__` or `__slots__` (etc.),
+/// that can be inserted into the
+/// source code as a `range_replacement` autofix.
+pub(super) fn sort_single_line_elements_sequence<F>(
+    kind: &SequenceKind,
+    elts: &[ast::Expr],
+    elements: &[&str],
+    locator: &Locator,
+    mut cmp_fn: F,
+) -> String
+where
+    F: FnMut(&str, &str) -> Ordering,
+{
+    assert_eq!(elts.len(), elements.len());
+    let (opening_paren, closing_paren) = kind.surrounding_parens(locator.contents());
+    let last_item_index = elements.len().saturating_sub(1);
+    let mut result = String::from(opening_paren);
+
+    let mut element_pairs = elements.iter().zip(elts).collect_vec();
+    element_pairs.sort_by(|(elem1, _), (elem2, _)| cmp_fn(elem1, elem2));
+    // We grab the original source-code ranges using `locator.slice()`
+    // rather than using the expression generator, as this approach allows
+    // us to easily preserve stylistic choices in the original source code
+    // such as whether double or single quotes were used.
+    for (i, (_, elt)) in element_pairs.iter().enumerate() {
+        result.push_str(locator.slice(elt));
+        if i < last_item_index {
+            result.push_str(", ");
+        }
+    }
+
+    result.push_str(closing_paren);
+    result
+}
+
+/// Create a string representing a fixed-up single-line
+/// definition of `__all__` or `__slots__` (etc.),
+/// that can be inserted into the
+/// source code as a `range_replacement` autofix.
+pub(super) fn sort_single_line_elements_dict<F>(
+    key_elts: &[ast::Expr],
+    elements: &[&str],
+    value_elts: &[ast::Expr],
+    locator: &Locator,
+    mut cmp_fn: F,
+) -> String
+where
+    F: FnMut(&str, &str) -> Ordering,
+{
+    assert!(key_elts.len() == elements.len() && elements.len() == value_elts.len());
```

Comment:
Nit: Splitting the assertion into two has the benefit that it generates more useful error messages (especially when using `assert_eq`), when it triggers (it otherwise leaves me wondering if the key length or values length is incorrect)